src/System.Private.ServiceModel/src/Internals/System/Runtime/IOThreadScheduler.cs

```diff
@@ -0,0 +1,664 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Contracts;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Runtime
+{
+    internal class IOThreadScheduler
```

Comment:
>IOThreadScheduler [](start = 19, length = 17)

Could you give us a quick overview of where this code came from and what parts are verbatim copies?  How much should I actually review?